doc/classes/Dictionary.xml

```diff
@@ -4,16 +4,60 @@
 		Dictionary type.
 	</brief_description>
 	<description>
-		Dictionary type. Associative container which contains values referenced by unique keys. Dictionaries are always passed by reference.
+		Dictionary type. Associative container which contains values referenced by unique keys. Just like in real life each dictionary composes multiple pairs of unique keys and values. Each pair is created by joining them via colon [code]Key:Value[/code].
 		Erasing elements while iterating over them [b]is not supported[/b].
 		Creating a dictionary:
 		[codeblock]
-		var d = {4: 5, "A key": "A value", 28: [1, 2, 3]}
+		var my_dir = {} # Creates an empty dictionary
+		var points_dir = {"White" : 50,"Yellow" : 75, "Orange" : 100}
+		[/codeblock]
+		You can access values of a dictionary by refrencing approrpiate key in above example [code]points_dir["White"][/code] would return value of 50.
+		[codeblock]
+		export(String, "White","Yellow", "Orange") var my_color
+		var points_dir = {"White" : 50,"Yellow" : 75, "Orange" : 100}
+		
+		func _ready():
+		    var points = points_dir[my_color]
+		[/codeblock]
+		In above code [code]var points[/code] will be assigned value that is paired with appropriate color selected in [code]var my_color[/code]
+		Dictionaries can contain more complex data 
+		[codeblock]
+		my_dir = {"First Array" : [1,2,3,4]} # Assigns Array to a Key
 		[/codeblock]
 		To add a key to an existing dictionary, access it like an existing key and assign to it:
 		[codeblock]
-		d[4] = "hello"  # Add integer 4 as a key and assign the String "hello" as its value.
-		d["Godot"] = 3.01  # Add String "Godot" as a key and assign the value 3.01 to it.
+		var points_dir = {"White" : 50,"Yellow" : 75, "Orange" : 100}
+		var points_dir["Blue"] = 150  # Add "Blue" as a key and assign 150 as its value.
+		[/codeblock]
+		Finally dictionaries can contain different types of keys and values in the same dictionary
+		[codeblock]
+		var my_dir = {"String Key" : 5, 4 : [1,2,3], 7 : "Hello"} # is a valid dictionary
+		[/codeblock]
+		Below methods allow you to perform more operations on dictionaries
+		[b]Note:[/b] that unlike Arrays you can't compare dictionaries directly
+		[codeblock]
+		array1 = [1, 2, 3]
+		array2 = [1, 2, 3]
+		
+		func compare_arrays():
+		    if array1 == array2:
+			    print("true") # Will print true
+			
+		dir1 = [a : 1, b : 2, c : 3]
+		dir2 = [a : 1, b : 2, c : 3]
+		
+		func compare_dictionaries():
+		    if dir1 == dir2:
+			    print("true") # Will NOT print true 
+		[/codeblock]
+		You need to first hash dictionary with [code].hash()[/code] method before you can compare them
+		[codeblock]
+		dir1 = [a : 1, b : 2, c : 3]
+		dir2 = [a : 1, b : 2, c : 3]
+		
+		func compare_compare_dictionaries():
+		    if dir1.hash() == dir2.hash():
+			    print("true") # Will print true 
```

Comment:
This will become obsolete once #29222 or a similar PR gets merged.

Also, two dictionaries with the same hashcode can have different contents, so this is not really a foul-proof way of checking equality.

(and finally, `[a : 1, b : 2]` is not a dictionary; `{a = 1, b = 2}`/`{"a": 1, "b": 2}` is)

Reply:
When PR is added we can update docs until then docs should reflect the actually engine not a future potential engine. 

I have fix the brackets and missing "